src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -160,7 +163,25 @@ private static PropagatedContingency create(Network network, Contingency conting
                     if (hvdcLine == null) {
                         throw new PowsyblException("HVDC line '" + element.getId() + "' not found in the network");
                     }
+                    HvdcAngleDroopActivePowerControl control = hvdcLine.getExtension(HvdcAngleDroopActivePowerControl.class);
+                    if (control != null && control.isEnabled() && hvdcAcEmulation) {
+                        throw new PowsyblException("Contingency on HVDC line '" + element.getId() + "' operated in AC emulation: not supported yet.");
+                    }
                     hvdcIdsToOpen.add(element.getId());
```

Comment:
For me:
- Use `hvdcIdsToOpen` only for HVDC in AC emulation, through a disabling of the line.
- Use converter station changes only for HVDC in active power setpoint.